inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,230 @@
+"""Mixin for file management.
+
+Methods to manage files and directories, within your local computer in
+order to prepare them for the simulation process. This mixin can be
+used to copy files and directories, and render template files with
+the given set of arguments.
+"""
+from typing import Union
+import pathlib
+import shutil
+import glob
+import io
+import os
+import re
+
+from absl import logging
+import jinja2
+
+from inductiva.utils import files, format_utils
+
+TEMPLATE_EXTENSION = ".jinja"
+SUFFIX_SEPARATOR = "__"
+
+
+class FileManager:
```

Comment:
Technically, it executes file management operations and can be used in general to manage files, however it is true that it encapsulates both just adding/copying files and rendering at the same time. We started with separate entities, both the code and structure of them was intrinsically similar so we decided to converge into one.

With this in mind, point 1) and 2) are just possible applications of this manager and we could also use it for other aspects, like moving files after the simulation finish into a single folder, or even manage the datasets from the simulation outputs. So I think the name is still appropriate for our context.

Reply:
> Technically, it executes file management operations and can be used in general to manage files, however it is true that it encapsulates both just adding/copying files and rendering at the same time. We started with separate entities, both the code and structure of them was intrinsically similar so we decided to converge into one.

I support this reasoning here. We had the `add_file|dir` and `render_file|dir` primitives and the end result was that everything was ver similar amongst the render/add groups.

As to having this mixin to manage outputs, I'm not sure it would be a valid use case for it.